grails-app/services/org/pih/warehouse/inventory/CycleCountProductAvailabilityService.groovy

```diff
@@ -124,7 +126,7 @@ class CycleCountProductAvailabilityService {
                     it.inventoryItem == cycleCountItem.inventoryItem &&
                     it.custom
         }
-        if (!wasItemEverCustomAdded && cycleCountItem.quantityCounted <= 0) {
+        if (!wasItemEverCustomAdded && (cycleCountItem.quantityCounted <= 0 || removeOutOfStockItemsImplicitly)) {
```

Comment:
since we have two collapsing behaviors - we want NOT to remove such an item when "zeroing" stock happens while count, but we want to remove the item when it happens while recount (reload, submitting the recount), my suggestion is to just add a flag determining whether we want to remove such an item implicitly even it quantityCounted is <= 0.
Behavior for custom items stays the same.